src/scripts/main.js

```diff
@@ -1 +1,109 @@
 'use strict';
+
+let rightClick;
+let leftClick;
+
+function makeDiv(className) {
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.setAttribute('class', className);
+  document.body.appendChild(div);
+
+  return div;
+}
+
+const firstPromise = () => {
+  return new Promise((resolve, reject) => {
+    document.addEventListener(
+      'click',
+      () => {
+        leftClick = true;
+
+        resolve('First promise was resolved');
+      },
+      { once: true },
+    );
+
+    // eslint-disable-next-line prefer-promise-reject-errors
+    setTimeout(() => reject('First promise was rejected!'), 3000);
+  });
```

Comment:
The 'firstPromise' function sets an event listener for clicks and a timeout to reject the promise. However, there is no cleanup for the event listener if the promise is rejected by the timeout. This can lead to memory leaks. Consider removing the event listener in the 'setTimeout' callback.